library/hkdf.c

```diff
@@ -136,6 +136,8 @@ int mbedtls_hkdf_expand( const mbedtls_md_info_t *md, const unsigned char *prk,
         goto exit;
     }
 
+    memset( t, 0, hash_len );
```

Comment:
I don't see any check in this function that hash_len <= MBEDTLS_MD_MAX_SIZE. Given the name of the macro this is probably true. Where is this enforced in the code (I looked a bit around but didn't find it)?

Reply:
`hash_len` is returned by `mbedtls_md_get_size`, which return the `size` field of a `mbedtls_md_info` data structure defined in `md.c`. The definition of `MBEDTLS_MD_MAX_SIZE` in `md.h` satisfies the property that it's at least as large as any size of a digest algorithm whose info structure is included in the build of `md.c`. We only enforce this by code inspection, but that's ok because new digest algorithms come up extremely rarely. (It would be good to test this in `test_suite_md`, similar to the tests in `test_suite_psa_crypto_metadata` which validate the PSA analog `PSA_HASH_SIZE(any_supported_hash) <= PSA_HASH_MAX_SIZE`.)

Reply:
ok thanks for the explanation.